photutils/psf/psfphotometry.py

```diff
@@ -0,0 +1,446 @@
+            the sources and ``flux`` contains flux estimates of the sources.
+            See, e.g., `~photutils.detection.DAOStarFinder`
+        group : callable or instance of any GroupStarsBase subclasses
+            ``group`` should be able to decide whether a given star overlaps
+            with any other and label them as beloging to the same group.
+            ``group`` receives as input an `~astropy.table.Table` object
+            with columns named as ``id``, ``x_0``, ``y_0``, in which ``x_0``
+            and ``y_0`` have the same meaning of ``xcentroid`` and
+            ``ycentroid``. This callable must return an `~astropy.table.Table`
+            with columns ``id``, ``x_0``, ``y_0``, and ``group_id``. The
+            column ``group_id`` should cotain integers starting from ``1``
+            that indicate in which group a given source belongs to.
+            See, e.g., `~photutils.psf.DAOGroup`
+        bkg : callable or instance of any `~photutils.BackgroundBase`
+            subclasses ``bkg`` should be able to compute either a scalar
+            background or a 2D background of a given 2D image.
+            See, e.g., `~photutils.background.MedianBackground`
+        psf : `astropy.modeling.Fittable2DModel` instance
+            PSF or PRF model to fit the data. Could be one of the models in
+            this package like `~photutils.psf.sandbox.DiscretePRF`,
+            `~photutils.psf.IntegratedGaussianPRF`, or any other suitable
+            2D model.
+            This object needs to identify three parameters (position of
+            center in x and y coordinates and the flux) in order to set them
+            to suitable starting values for each fit. The names of these
+            parameters can be given as follows:
+
+            - Set ``psf.psf_xname``, ``psf.psf_yname`` and
+              ``psf.psf_fluxname`` to strings with the names of the respective
+              psf model parameter.
+            - If those attributes are not found, the names ``x_0``, ``y_0``
+              and ``flux`` are assumed.
+
+            `~photutils.psf.prepare_psf_model` can be used to prepare any 2D
+            model to match these assumptions.
+        fitshape : array-like
+            Rectangular shape around the center of a star which will be used
+            to collect the data to do the fitting, e.g. (5, 5) means to take
+            the following relative pixel positions: [-2, -1, 0, 1, 2].
+            Also, each element of ``fitshape`` must be an odd number.
+        fitter : Fitter instance (default=LevMarLSQFitter())
+            Fitter object used to compute the optimized centroid positions
+            and/or flux of the identified sources. See
+            `~astropy.modeling.fitting` for more details on fitters.
+        niters : int (default=3)
+            Number of iterations to perform the loop FIND, GROUP, SUBTRACT,
+            NSTAR.
+        aperture_radius : float (default=None)
```

Comment:
I like the fact that the default is to try the fwhm, but it would be nice if one could specify *fractions* of FWHM's.  Maybe instead of using None, you can say that any value of `aperture_radius` less than 0 is a FWHM fraction?  So then the default could just be ``-1.0`` to get the same behavior? (and then my comment below about accounting for the None case can be ignored)